src/mmw/js/src/analyze/models.js

```diff
@@ -141,89 +141,210 @@ var AnalyzeTaskCollection = Backbone.Collection.extend({
+                    taskName: "analyze/terrain"
+                }
+            ]
         },
         {
             name: "climate",
             displayName: "Climate",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/climate",
-            enabledForCatalogMode: true,
+            tasks: [
+                {
+                    name: "climate",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/climate",
+                    enabledForCatalogMode: true,
+                },
+            ]
         },
         {
             name: "pointsource",
             displayName: "Pt Sources",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/pointsource"
+            tasks: [
+                {
+                    name: "pointsource",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/pointsource"
+                }
+            ]
         },
         {
             name: "animals",
             displayName: "Animals",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/animals"
+            tasks: [
+                {
+                    name: "animals",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/animals"
+                }
+            ]
         },
         {
             name: "catchment_water_quality",
             displayName: "Water Qual",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/catchment-water-quality"
-        },
-        {
-            name: "protected_lands",
-            displayName: "Protected Lands",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/protected-lands"
+            tasks: [
+                {
+                    name: "catchment_water_quality",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/catchment-water-quality"
+                }
+            ]
         },
     ];
 
     // Remove analyses which aren't available for catalog search mode
     if (settings.get('data_catalog_enabled')) {
-        tasks = _.filter(tasks, { enabledForCatalogMode: true });
+        var filteredTaskGroups = [];
+        _.foreach(taskGroups, function(tg) {
+            var filteredTasks = _.filter(tg.tasks, { enabledForCatalogMode: true });
+            if(filteredTasks.length > 0) {
+                tg.tasks = filteredTasks;
+                filteredTaskGroups.push(tg);
+            }
+        });
+
+        taskGroups = filteredTaskGroups;
     }
 
-    return new AnalyzeTaskCollection(tasks);
+    taskGroups = _.map(taskGroups, function (tg) {
+        tg.tasks = new AnalyzeTaskCollection(tg.tasks);
```

Comment:
Same here. Also, is there a way to not override `tg.tasks`? Maybe do the first pass on a different array?

Reply:
Sorry, not clear what you mean by override?

Reply:
I have a cleaner suggestion, closing this thread.